src/tests/stats/test_distribution_uniform.fypp

```diff
@@ -0,0 +1,282 @@
+#:include "common.fypp"
+#:set RC_KINDS_TYPES = REAL_KINDS_TYPES + CMPLX_KINDS_TYPES
+#:set ALL_KINDS_TYPES = INT_KINDS_TYPES + RC_KINDS_TYPES
+program test_distribution_uniform
+    use stdlib_error, only : check
+    use stdlib_kinds
+    use stdlib_stats_distribution_PRNG, only : random_seed, dist_rand
+    use stdlib_stats_distribution_uniform,                                      &
+        uni_rvs => uniform_distribution_rvs,                                    &
+        uni_pdf => uniform_distribution_pdf,                                    &
+        uni_cdf => uniform_distribution_cdf
+
+    implicit none
+    logical ::  warn = .true.
+    #:for k1, t1 in REAL_KINDS_TYPES
+    ${t1}$, parameter :: ${k1}$tol = 1000 * epsilon(1.0_${k1}$)
+    #:endfor
+    integer :: put, get
+
+    put = 135792468
+
+    call test_shuffle
+
+    call test_uni_rvs_0
+
+    #:for k1, t1 in ALL_KINDS_TYPES
+    call test_uni_rvs_${t1[0]}$${k1}$
+    #:endfor
+
+    #:for k1, t1 in ALL_KINDS_TYPES
+    call test_uni_pdf_${t1[0]}$${k1}$
+    #:endfor
+
+    #:for k1, t1 in ALL_KINDS_TYPES
+    call test_uni_cdf_${t1[0]}$${k1}$
+    #:endfor
+    stop
+
+    contains
+
+    subroutine test_shuffle
+        integer :: n(10), na(10) = [10, 6, 9, 2, 8, 1, 3, 5, 7, 4]
+        real :: x(10),  xa(10)= [5.0, 10.0, 9.0, 4.0, 3.0, 8.0, 2.0, 1.0,       &
+                                 7.0, 6.0]
+        complex :: z(10), za(10)=[(8.0, 8.0), (7.0, 7.0), (4.0, 4.0),           &
+                                  (1.0, 1.0), (5.0, 5.0), (9.0, 9.0),           &
+                                  (6.0, 6.0), (3.0, 3.0), (2.0, 2.0),           &
+                                  (10.0, 10.0)]
+        integer :: i, put, get
+
+        do i=1, 10
+            n(i) = i
+            x(i) = real(i)
+            z(i) = cmplx(real(i),real(i))
+        end do
+        put = 32165498
+        call random_seed(put, get)
+        n(:) = shuffle(n)
+        x(:) = shuffle(x)
+        z(:) = shuffle(z)
+        call check(all(n == na), &
+             msg="Integer shuffle failed test", warn=warn)
```

Comment:
Since the seed is always the same, the output of shuffle should be always the same too, right?
If yes, then I think it would be good to test against the expected output of shuffle

Reply:
I don't quite understand your point. It is tested against the output from shuffle.

Reply:
Sorry. I missread the test.